server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java

```diff
@@ -134,6 +138,18 @@ public void setLocations(Collection<Assignment> assignments) throws DistributedS
 
     TabletMutator tabletMutator = ample.mutateTablet(assignment.tablet);
     tabletMutator.putLocation(assignment.server, LocationType.CURRENT);
+    if ("assignment".equals(context.getConfiguration().get(Property.TSERV_LAST_LOCATION_MODE))) {
+      TabletMetadata lastMetadata =
+          ample.readTablet(assignment.tablet, TabletMetadata.ColumnType.LAST);
+      if (lastMetadata != null && lastMetadata.getLast() != null) {
+        if (!lastMetadata.getLast().equals(assignment.server)) {
+          tabletMutator.putLocation(assignment.server, LocationType.LAST);
+          tabletMutator.deleteLocation(lastMetadata.getLast(), LocationType.LAST);
+        }
+      } else {
+        tabletMutator.putLocation(assignment.server, LocationType.LAST);
+      }
+    }
```

Comment:
I wonder if this whole block can be put into a method that's common to both state stores. They differ very little.

Reply:
That I can do.  I will work that today.

Reply:
I refactored the location updating into methods in the ManagerMetadataUtil class.  If you want them located elsewhere or in their own class just let me know.